documentation/docs/tutorials/tutorial.md

```diff
@@ -0,0 +1,596 @@
+        "dist": "tsc",
+        "start": "node dist/index"
+    },
+    "dependencies": {
+        "@iota/crypto.js": "2.0.0-rc.1",
+        "@iota/iota.js": "2.0.0-rc.1",
+        "@iota/util.js": "2.0.0-rc.1",
+        "@iota/pow-neon.js": "2.0.0-rc.2"
+    },
+    "devDependencies": {
+        "typescript": "^4.4.3",
+        "@types/node": "18.7.23"
+    }
+}
+```
+
+## Generating addresses
+
+In order to start transacting in the network it is needed to count with one or more addresses that can unlock outputs or receive new outputs. Addresses are derived from a public key and their associated outputs can be unlocked with the corresponding private key. As it happens with Firefly wallets, the idea is to be able to generate multiple addresses starting with an initial master secret (*seed phrase*), and, from that point on, use a hierarchical deterministic method to derive multiple addresses.
+
+### Generating a seed phrase and an Ed25519 seed
+
+The seed phrase is generated in accordance with the [BIP39](https://github.com/bitcoin/bips/blob/master/bip-0039.mediawiki) specification and it is composed by a set of words, represented as a whitespace-separated string.
+
+```typescript
+// Default entropy length is 256
+const randomMnemonic = Bip39.randomMnemonic();
+
+console.log("Seed phrase:", randomMnemonic);
+```
+
+Once you have the seed phrase (aka BIP39 random mnemonic) the next step is to obtain a *Ed25519 master seed* from the seed phrase:
+
+```typescript
+const masterSeed = Ed25519Seed.fromMnemonic(randomMnemonic);
+```
+
+This Ed25519 master seed will be used later to generate as many Ed25519 key pairs as needed through the BIP32 deterministic method.
+
+### Deterministic address paths (BIP32)
+
+The aforementioned master seed can be used to generate, in a deterministic manner, addresses i.e. Ed25519 key pairs, generated through the [BIP32](https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki) method and structured as per the [BIP44](https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki) logical hierarchy. The `iota.js` library provides a method `generateBip44Address` that creates these BIP32 paths using an state object that it is updated on each call made. See example below
+
+```typescript
+const NUM_ADDR = 6;
+const addressGeneratorAccountState = {
+    accountIndex: 0,
+    addressIndex: 0,
+    isInternal: false
+};
+const paths: string[] = [];
+for (let i = 0; i < NUM_ADDR; i++) {
+    const path = generateBip44Address(addressGeneratorAccountState);
+    paths.put(path);
+
+    console.log(`${path}`);
+}
+```
+
+That will generate the following BIP32 paths:
+
+```text
+m/44'/4218'/0'/0'/0'
+m/44'/4218'/0'/1'/0'
+m/44'/4218'/0'/0'/1'
+m/44'/4218'/0'/1'/1'
+m/44'/4218'/0'/0'/2'
+m/44'/4218'/0'/1'/2'
+```
+
+* `44` is a constant that denotes *purpose*, (`0x8000002C`) following the [BIP43](https://github.com/bitcoin/bips/blob/master/bip-0043.mediawiki) recommendation.
+* `4218` is the *coin type* defined for Shimmer
```

Comment:
The `generateBip44Address` has been updated recently, so that now it also takes an _optional_ `COIN_TYPE` parameter. This lets lib users change the coin type they want to generate for. Also a new constant value has been added so that `4218` is for `iota` and `4219` is for shimmer.
Visible here https://github.com/iotaledger/iota.js/blob/feat/stardust/packages/iota/src/highLevel/addresses.ts